ako-gateway-api/nodes/dequeue_ingestion.go

```diff
@@ -1 +1,113 @@
+/*
+ * Copyright 2023-2024 VMware, Inc.
+ * All Rights Reserved.
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*   http://www.apache.org/licenses/LICENSE-2.0
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
 package nodes
+
+import (
+	akogatewayapilib "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/lib"
+	akogatewayapiobjects "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/ako-gateway-api/objects"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/lib"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/nodes"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/internal/objects"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+func DequeueIngestion(key string, fullsync bool) {
+	objType, namespace, name := lib.ExtractTypeNameNamespace(key)
+
+	schema, valid := ConfigDescriptor().GetByType(objType)
+	if !valid {
+		return
+	}
+	gatewayList, found := schema.GetGateways(name, namespace, key)
+	if !found {
+		//returning due to error, cannot delete or update
+		return
+	}
+	handleGateways(gatewayList, fullsync, key)
+}
+
+func handleGateways(gatewayList []string, fullsync bool, key string) {
+	sharedQueue := utils.SharedWorkQueue().GetQueueByName(utils.GraphLayer)
+	for _, gateway := range gatewayList {
+		_, namespace, name := lib.ExtractTypeNameNamespace(gateway)
+
+		modelName := lib.GetModelName(lib.GetTenant(), akogatewayapilib.GetGatewayParentName(namespace, name))
+		modelFound, _ := objects.SharedAviGraphLister().Get(modelName)
```

Comment:
we need to revisit the Get model and create new model logic.

Reply:
will be updated with httproute graph layer changes